### PR TITLE
Optimality calculator

### PR DESCRIPTION
**Pull request by jakubgros**

### CODE REVIEW

File: Back-end/src/com/smartplanner/OptimalityCalculator.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        this.optimizedActivity = optimizedActivity;
+        this.timeDistanceManager = timeDistanceManager;
+        this.maxCommutesPerDay = maxCommutesPerDay;
+        this.minTimeSpentOnOptimizedAtOnceInMinutes = minTimeSpentOnOptimizedAtOnceInMinutes;
+        this.numberOfDaysInCycle = numberOfDaysInCycle;
+    }
+
+    public OptimalTimetable calculate(ArrayList<TimetableEntry> timetable){
+        ArrayList<ArrayList<Boolean>> optimalDecisionPoints = new ArrayList<ArrayList<Boolean>>(numberOfDaysInCycle);
+        int timeSpentInWorkInCycle=0;
```
Comment (kucharzyk-sebastian) [inline Back-end/src/com/smartplanner/OptimalityCalculator.java]: Missing spaces around =

File: Back-end/src/com/smartplanner/OptimalTimetable.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    // first diemension is responsible for the day in cycle
+    // second dimension is responsible for decision point
+    // i.e.
+    // optimalDecisionPoints.get(0).get(0) answers question: should I go to work in day 0 before first
+    // optimalDecisionPoints.get(0).get(1) answers question: should I go to work in day 0 after first activity
+    // ...
+    // optimalDecisionPoints.get(0).get(n) answers question: should I go to work in day 0 after last activity (where n = last index in array)
+
+
+    public OptimalTimetable(int minutesSpentAtOptimizedActivity, ArrayList<TimetableEntry> optimalTimetable, ArrayList<ArrayList<Boolean>> optimalDecisionPoints) {
```
Comment (nowakkamil) [inline Back-end/src/com/smartplanner/OptimalTimetable.java]: Please consider splitting this line of code.

File: Back-end/src/com/smartplanner/OptimalityCalculator.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    private int calculateRegardingProvidedDecisionPoints(ArrayList<TimetableEntry> timetable, ArrayList<Boolean> currDecisionPoints, int cycleDayNumber) {
+        if (getNumberOfOnes(currDecisionPoints) > maxCommutesPerDay) //current case imposes more commutes per day than maximal allowed
+            return 0;
+
+        int timeSpentInWork = 0;
+        ArrayList<TimetableEntry> specifiedDayTimetable = extractEntriesForSpecifiedDay(timetable, cycleDayNumber);
+
+        if (currDecisionPoints.get(0) == true) {
+            int timeBetweenOptimizedActivityOpenAndFirstActivityStart = calculateMinutesBetweenTwoTimePoints(optimizedActivity.getOpensAt(),
+                    specifiedDayTimetable.get(0).getTerm().getStartTime()) - timeDistanceManager.getTimeDistanceInMin(specifiedDayTimetable.get(0).getActivity(), optimizedActivity);
```
Comment (nowakkamil) [inline Back-end/src/com/smartplanner/OptimalityCalculator.java]: Please consider splitting this line of code.

File: Back-end/src/com/smartplanner/OptimalityCalculator.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    private int getNumberOfOnes(ArrayList<Boolean> decisionPoints) {
+        int numberOfOnes = 0;
+        for (Boolean decision : decisionPoints)
+            if (decision == true)
+                ++numberOfOnes;
+
+        return numberOfOnes;
+    }
+
```
Comment (nowakkamil) [inline Back-end/src/com/smartplanner/OptimalityCalculator.java]: Could you break the lines in this method where the single line width exceeds 120 characters?

**Pull request closed without merging**